exonum-java-binding-core/src/test/java/com/exonum/binding/blockchain/BlockchainTest.java

```diff
@@ -128,7 +129,18 @@ void getTxResult() {
     when(mockMapIndex.get(messageHash)).thenReturn(txResult);
     when(mockSchema.getTxResults()).thenReturn(mockMapIndex);
 
-    assertThat(blockchain.getTxResult(messageHash)).isEqualTo(txResult);
+    assertThat(blockchain.getTxResult(messageHash).get()).isEqualTo(txResult);
+  }
+
+  @Test
+  void getNonexistentTxResult() {
+    ProofMapIndexProxy mockMapIndex = mock(ProofMapIndexProxy.class);
+    HashCode messageHash = HashCode.fromString("ab");
+
+    when(mockMapIndex.get(messageHash)).thenReturn(null);
+    when(mockSchema.getTxResults()).thenReturn(mockMapIndex);
+
+    assertThat(blockchain.getTxResult(messageHash)).isEqualTo(Optional.empty());
```

Comment:
You can use (here and elsewhere) https://joel-costigliola.github.io/assertj/core-8/api/org/assertj/core/api/AbstractOptionalAssert.html#isEmpty--